tools/Vrops.py

```diff
@@ -187,17 +185,23 @@ def get_resources(self, target: str,
             return resources, response.status_code
 
     def get_datacenter(self, target, token, parent_uuids, query_specs):
-        return self.get_resources(target, token, adapterkind="VMWARE", resourcekinds=["Datacenter"],
-                                  uuids=parent_uuids, query_specs=query_specs)
+        resourcekind = 'Datacenter'
+        q_specs = query_specs.get(resourcekind) if resourcekind in query_specs else query_specs.get('default', {})
+        return self.get_resources(target, token, adapterkind="VMWARE", resourcekinds=[resourcekind],
+                                  uuids=parent_uuids, query_specs=q_specs)
```

Comment:
I like this behaviour more now.
We are introducing some code duplication here though.

Why aren't we doing this in a dedicated function and are calling this function at the beginning of `get_resources`?

Pretty much the same behaviour, we're passing in `resourcekinds` and are giving back the `q_specs`. We then can handle it centrally.

Reply:
I know. This is not as DRY as I would like it to be. But there is a tradeoff. 

Inside `get_resources` is not the best way. The complexity starts with a list of `resourcekinds` like `sddc` and `vcops`.  They are always requested with default specifications. But we add more complexity to this function when we check each `resourcekind` inside the function. Let us do it outside the function. I have proposed a new commit.

Reply:
That's the spirit, like it :)